### PR TITLE
Create RoboAppCompatActivity and updated appcompat dependency version

### PR DESCRIPTION
**Pull request by felipeleusin**
This is basically a clone of RoboActionBarActivity using AppCompatActivity instead and updated the dependency on appcompat-v7.

Fixes issue #298

### CODE REVIEW

**Comment by nicolasjafelle** [comment]
+1

**Comment by fedeperez89** [comment]
+1

File: pom.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
 		<!--LIBRARIES -->
 		<android-map.version>7_r1</android-map.version>
 		<android-support-v4.version>19</android-support-v4.version>
-		<android-appcompat-v7.version>19</android-appcompat-v7.version>
+		<android-appcompat-v7.version>22.2.0</android-appcompat-v7.version>
```
Comment (bbarnhill) [inline pom.xml]: this can be updated to 23 now

**Comment by bbarnhill** [comment]
+1 we really need this badly

**Comment by felipeleusin** [comment]
No idea. I was just testing it but went to use separate libraries instead. With ButterKnife + Dagger2 you can get pretty much the same features I was looking for here

**Comment by fedeperez89** [comment]
@bbarnhill  There is a issue asking if it was still mantained #311

**Comment by stephanenicolas** [comment]
Why can't you upgrade the dependency in your own project ?
Maven will use this one instead of the older version.

2015-09-02 22:43 GMT-04:00 Brad Barnhill <notifications@github.com>:

**Comment by bbarnhill** [comment]
Because if you inherit from RoboActivity you cant also inherit from AppCompatActivity (Multiple Inheritence) ... is that what you were talking about?

**Comment by stephanenicolas** [comment]
More or less.

Our point is that we basically duplicate Robo* classes, only the base class
changes which is not a big win for us.
So we will let this to the developers starting from next release. We will
clearly explain how to create a base class that suits your apps needs, but
won't provide one for every possibility of activity / fragment inheritance.


2015-09-08 10:47 GMT-04:00 Brad Barnhill <notifications@github.com>:

**Comment by bbarnhill** [comment]
right that was my understanding from reading on the next release (comments from others) ... thats how we are going to tackle this in the short term as well.

**Comment by brindy** [comment]
+1 for this.  I basically only ever use AppCompatActivity these days and always have to create a custom BaseActivity just for the RoboGuice stuff.

**Comment by davidlad123** [comment]
+1 for this, I'm about to integrate it into a project. If this works without issues, i'll let everyone know.

**Comment by Gunmer** [comment]
+1

**Comment by oc243** [comment]
+1

**Comment by barnhill** [comment]
This is the main reason we have pulled roboguice from our projects.

**Comment by ajaykumarmeher** [comment]
If anybody still need support for AppCompatActivity can try this below code,

import android.annotation.TargetApi;
import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.AttributeSet;
import android.view.View;

import com.google.inject.Inject;
import com.google.inject.Key;
import com.google.inject.internal.util.Stopwatch;

import java.lang.reflect.Constructor;
import java.util.HashMap;
import java.util.Map;

import roboguice.RoboGuice;
import roboguice.activity.event.OnActivityResultEvent;
import roboguice.activity.event.OnContentChangedEvent;
import roboguice.activity.event.OnNewIntentEvent;
import roboguice.activity.event.OnPauseEvent;
import roboguice.activity.event.OnRestartEvent;
import roboguice.activity.event.OnResumeEvent;
import roboguice.activity.event.OnSaveInstanceStateEvent;
import roboguice.activity.event.OnStopEvent;
import roboguice.context.event.OnConfigurationChangedEvent;
import roboguice.context.event.OnCreateEvent;
import roboguice.context.event.OnDestroyEvent;
import roboguice.context.event.OnStartEvent;
import roboguice.event.EventManager;
import roboguice.inject.ContentViewListener;
import roboguice.inject.RoboInjector;
import roboguice.util.RoboContext;

/**
- Created by ajayk on 8/20/2016.
  */
  public class RoboAppCompatActivity extends AppCompatActivity implements RoboContext {
  protected EventManager eventManager;
  protected HashMap<Key<?>,Object> scopedObjects = new HashMap<Key<?>, Object>();
  
  @Inject
  ContentViewListener ignored; // BUG find a better place to put this
  private Stopwatch stopwatch;
  
  @Override
  protected void onCreate(Bundle savedInstanceState) {
      stopwatch = new Stopwatch();
      final RoboInjector injector = RoboGuice.getInjector(this);
      stopwatch.resetAndLog("RoboActivity creation of injector");
      eventManager = injector.getInstance(EventManager.class);
      stopwatch.resetAndLog("RoboActivity creation of eventmanager");
      injector.injectMembersWithoutViews(this);
      stopwatch.resetAndLog("RoboActivity inject members without views");
      super.onCreate(savedInstanceState);
      stopwatch.resetAndLog("RoboActivity super onCreate");
      eventManager.fire(new OnCreateEvent<Activity>(this,savedInstanceState));
      stopwatch.resetAndLog("RoboActivity fire event");
  }
  
  @Override
  protected void onSaveInstanceState(Bundle outState) {
      super.onSaveInstanceState(outState);
      eventManager.fire(new OnSaveInstanceStateEvent(this, outState));
  }
  
  @Override
  protected void onRestart() {
      super.onRestart();
      eventManager.fire(new OnRestartEvent(this));
  }
  
  @Override
  protected void onStart() {
      super.onStart();
      eventManager.fire(new OnStartEvent<Activity>(this));
  }
  
  @Override
  protected void onResume() {
      super.onResume();
      eventManager.fire(new OnResumeEvent(this));
  }
  
  @Override
  protected void onPause() {
      super.onPause();
      eventManager.fire(new OnPauseEvent(this));
  }
  
  @Override
  protected void onNewIntent( Intent intent ) {
      super.onNewIntent(intent);
      eventManager.fire(new OnNewIntentEvent(this));
  }
  
  @Override
  protected void onStop() {
      try {
          eventManager.fire(new OnStopEvent(this));
      } finally {
          super.onStop();
      }
  }
  
  @Override
  protected void onDestroy() {
      try {
          eventManager.fire(new OnDestroyEvent<Activity>(this));
      } finally {
          try {
              RoboGuice.destroyInjector(this);
          } finally {
              super.onDestroy();
          }
      }
  }
  
  @Override
  public void onConfigurationChanged(Configuration newConfig) {
      final Configuration currentConfig = getResources().getConfiguration();
      super.onConfigurationChanged(newConfig);
      eventManager.fire(new OnConfigurationChangedEvent<Activity>(this,currentConfig, newConfig));
  }
  
  @Override
  public void onContentChanged() {
      super.onContentChanged();
      RoboGuice.getInjector(this).injectViewMembers(this);
      eventManager.fire(new OnContentChangedEvent(this));
  }
  
  @Override
  protected void onActivityResult(int requestCode, int resultCode, Intent data) {
      super.onActivityResult(requestCode, resultCode, data);
      eventManager.fire(new OnActivityResultEvent(this, requestCode, resultCode, data));
  }
  
  @Override
  public Map<Key<?>, Object> getScopedObjectMap() {
      return scopedObjects;
  }
  
  @Override
  public View onCreateView(String name, Context context, AttributeSet attrs) {
      if (RoboAppCompatActivity.shouldInjectOnCreateView(name))
          return RoboAppCompatActivity.injectOnCreateView(name, context, attrs);
  
  ```
  return super.onCreateView(name, context, attrs);
  ```
  
  }
  
  @Override
  @TargetApi(Build.VERSION_CODES.HONEYCOMB)
  public View onCreateView(View parent, String name, Context context, AttributeSet attrs) {
      if (RoboAppCompatActivity.shouldInjectOnCreateView(name))
          return RoboAppCompatActivity.injectOnCreateView(name, context, attrs);
  
  ```
  return super.onCreateView(parent, name, context, attrs);
  ```
  
  }
  
  /**
  - @return true if name begins with a lowercase character (indicating a package) and it doesn't start with com.android
    */
    protected static boolean shouldInjectOnCreateView(String name) {
    return false; // && Character.isLowerCase(name.charAt(0)) && !name.startsWith("com.android") && !name.equals("fragment");
    }
  
  protected static View injectOnCreateView(String name, Context context, AttributeSet attrs) {
      try {
          final Constructor<?> constructor = Class.forName(name).getConstructor(Context.class, AttributeSet.class);
          final View view = (View) constructor.newInstance(context, attrs);
          RoboGuice.getInjector(context).injectMembers(view);
          RoboGuice.getInjector(context).injectViewMembers(view);
          return view;
      } catch (Exception e) {
          throw new RuntimeException(e);
      }
  }

}

**Pull request closed without merging**